test/export/InstanceExporter.test.ts

```diff
@@ -0,0 +1,94 @@
+    input = new FSHTank([doc], { name: 'test', version: '0.0.1', canonical: 'http://example.com' });
+    exporter = new InstanceExporter(defs, input);
+  });
+
+  it('should output empty results with empty input', () => {
+    const exported = exporter.export();
+    expect(exported).toEqual([]);
+  });
+
+  it('should export a single instance', () => {
+    const instance = new Instance('MyInstance');
+    instance.instanceOf = 'Patient';
+    doc.instances.set(instance.name, instance);
+    const exported = exporter.export();
+    expect(exported.length).toBe(1);
+  });
+
+  it('should export multiple instances', () => {
+    const instanceFoo = new Instance('Foo');
+    instanceFoo.instanceOf = 'Patient';
+    const instanceBar = new Instance('Bar');
+    instanceBar.instanceOf = 'Patient';
+    doc.instances.set(instanceFoo.name, instanceFoo);
+    doc.instances.set(instanceBar.name, instanceBar);
+    const exported = exporter.export();
+    expect(exported.length).toBe(2);
+  });
+
+  it('should still export instance if one fails', () => {
+    const instanceFoo = new Instance('Foo');
+    instanceFoo.instanceOf = 'Baz';
+    const instanceBar = new Instance('Bar');
+    instanceBar.instanceOf = 'Patient';
+    doc.instances.set(instanceFoo.name, instanceFoo);
+    doc.instances.set(instanceBar.name, instanceBar);
+    const exported = exporter.export();
+    expect(exported.length).toBe(1);
+    expect(exported[0].instanceName).toBe('Bar');
+  });
+
+  it('should log a message with source information when the parent is not found', () => {
+    const instance = new Instance('Bogus').withFile('Bogus.fsh').withLocation([2, 9, 4, 23]);
+    instance.instanceOf = 'BogusParent';
+    doc.instances.set(instance.name, instance);
+    exporter.export();
+    expect(loggerSpy.getLastMessage()).toMatch(/File: Bogus\.fsh.*Line: 2 - 4\D/s);
+  });
+
+  it('should export instances with InstanceOf FSHy profile', () => {
+    const profileFoo = new Profile('Foo');
+    profileFoo.parent = 'Patient';
+    const instanceBar = new Instance('Bar');
+    instanceBar.instanceOf = 'Foo';
+    doc.profiles.set(profileFoo.name, profileFoo);
+    doc.instances.set(instanceBar.name, instanceBar);
+    const exported = exporter.export();
+    expect(exported.length).toBe(1); // One instance is successfully exported because profile is defined
+    expect(exported[0].instanceName).toBe('Bar');
+    expect(exported[0].resourceType).toBe('Patient');
+  });
+
+  it('should fix values on an instance', () => {
+    const instance = new Instance('Bar');
+    instance.instanceOf = 'Patient';
+    const fixedValRule = new FixedValueRule('gender');
+    const fixedFshCode = new FshCode('foo', 'http://foo.com');
+    fixedValRule.fixedValue = fixedFshCode;
+    instance.rules.push(fixedValRule);
+    doc.instances.set(instance.name, instance);
+    const exported = exporter.export();
+    expect(exported.length).toBe(1);
+    // @ts-ignore
```

Comment:
I think if you add a line to `InstanceDefinition` like this: `[key: string]: any`, that indicates that any key value pair is allowed on `InstanceDefinition`, so the `ts-ignore` wouldn't be needed. In my mind adding this line would make sense, since an Instance can have any key value pair defined by FHIR. Of which there are so many that trying to list them all out would be a bad idea. But I'm not confident enough in my TypeScript best practices to know if `[key: string]: any` is a bad idea. @mint-thompson, do you have any thoughts?

Reply:
I think `[key: string]: any` makes sense in this very specific context.